src/app/core/services/schedule/schedule.service.ts

```diff
@@ -86,20 +93,24 @@ export class ScheduleService {
     })
   }
 
-  setTasks(type: TaskType, tasks): Promise<void> {
+  setTasks(type: AssessmentType, tasks): Promise<void> {
     switch (type) {
-      case TaskType.NON_CLINICAL:
-        return this.setNonClinicalTasks(tasks)
-      case TaskType.CLINICAL:
-        return this.setClinicalTasks(tasks)
+      case AssessmentType.SCHEDULED:
+        return this.setScheduledTasks(tasks)
+      case AssessmentType.ON_DEMAND:
+        return this.setOnDemandTasks(tasks)
```

Comment:
I am sorry i don't understand, are we replacing clinical tasks with on demand? I thought we were creating a new onDemand task and keeping the clinical functionality as it is.

Reply:
Sorry @yatharthranjan, these were mistakes in modifying on-demand questionnaires to be an additional questionnaire (as opposed to replacing clinical tasks). I got confused with all the changes. 😅 I've corrected them now.